pd/src/verify/stateless.rs

```diff
@@ -99,23 +99,36 @@ impl StatelessTransactionExt for Transaction {
                     delegations.push(delegate);
                 }
                 Action::Undelegate(undelegate) => {
-                    // There are currently no stateless verification checks than the ones implied by
-                    // the binding signature.
-                    undelegations.push(undelegate);
+                    if undelegation.is_none() {
+                        undelegation = Some(undelegate);
+                    } else {
+                        return Err(anyhow::anyhow!("Multiple undelegations in one transaction"));
+                    }
                 }
                 _ => {
                     return Err(anyhow::anyhow!("unsupported action"));
                 }
             }
         }
 
+        // We prohibit actions other than `Spend`, `Delegate`, `Output` and `Undelegate` in
+        // transactions that contain `Undelegate`, to avoid having to quarantine them.
+        if undelegation.is_some() {
+            use Action::*;
+            for action in self.transaction_body().actions {
+                if !matches!(action, Undelegate(_) | Delegate(_) | Spend(_) | Output(_)) {
+                    return Err(anyhow::anyhow!("transaction contains an undelegation, but also contains an action other than Spend, Delegate, Output or Undelegate"));
+                }
+            }
+        }
```

Comment:
Additionally, we prohibit as a consensus rule any actions from sharing a transaction that contains an `Undelegate` other than `Delegate` (to allow redelegation), `Spend`, and `Output`. This means we don't need to handle the logic for things like "what if someone's swap gets quarantined" or "do quarantined votes count, and if so, when?"